src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HostFxrResolver.cpp

```diff
@@ -539,6 +531,60 @@ HostFxrResolver::InvokeWhereToFindDotnet()
     return result;
 }
 
+BOOL HostFxrResolver::IsX64(const WCHAR* dotnetPath)
+{
+    std::ifstream file(dotnetPath, std::ios::binary);
+    if (!file.is_open())
+    {
+        LOG_INFOF(L"Failed to open file %ls", dotnetPath);
+        return false;
+    }
+
+    // Read the DOS header
+    IMAGE_DOS_HEADER dosHeader{};
+    file.read(reinterpret_cast<char*>(&dosHeader), sizeof(dosHeader));
+    if (dosHeader.e_magic != IMAGE_DOS_SIGNATURE) // 'MZ'
+    {
+        LOG_INFOF(L"%ls is not a valid executable file (missing MZ header).", dotnetPath);
+        return false;
+    }
+
+    // Seek to the PE header
+    file.seekg(dosHeader.e_lfanew, std::ios::beg);
+
+    // Read the PE header
+    DWORD peSignature{};
+    file.read(reinterpret_cast<char*>(&peSignature), sizeof(peSignature));
+    if (peSignature != IMAGE_NT_SIGNATURE) // 'PE\0\0'
+    {
+        LOG_INFOF(L"%ls is not a valid PE file (missing PE header).", dotnetPath);
```

Comment:
Assuming INFO verbosity is the same as it is in .NET, is this the right verbosity to use? Maybe Trace?